### PR TITLE
Add support float types

### PR DESCRIPTION
**Pull request by janusheide**

### CODE REVIEW

File: test/src/test_big_endian.cpp [hunk 1]
```diff
@@ -208,4 +208,207 @@ TEST(test_big_endian, convert)
         out = endian::big_endian::get_bytes<sizeof(out), decltype(out)>(data);
         EXPECT_EQ(input, out);
     }
+
+    // Test float
+    {
+        uint8_t data[4];
+        float input = 0x0;
```
Comment (petya2164) [inline test/src/test_big_endian.cpp]: it is odd to use the hex notation for float ;)

File: test/src/test_big_endian.cpp [hunk 2]
```diff
@@ -208,4 +208,207 @@ TEST(test_big_endian, convert)
...
+        endian::big_endian::get_bytes<4>(out, data);
+        EXPECT_EQ(input, out);
+
+        out = endian::big_endian::get<decltype(out)>(data);
+        EXPECT_EQ(input, out);
+
+        out = endian::big_endian::get_bytes<sizeof(out), decltype(out)>(data);
+        EXPECT_EQ(input, out);
+
+        input = 0x1;
```
Comment (petya2164) [inline test/src/test_big_endian.cpp]: 1.0f

File: test/src/test_big_endian.cpp [hunk 3]
```diff
@@ -208,4 +208,207 @@ TEST(test_big_endian, convert)
...
+        EXPECT_EQ(0x3FU, data[0]);
+        EXPECT_EQ(0x80U, data[1]);
+        EXPECT_EQ(0x00U, data[2]);
+        EXPECT_EQ(0x00U, data[3]);
+
+        out = 0;
+        endian::big_endian::get_bytes<4>(out, data);
+        EXPECT_EQ(input, out);
+
+        out = endian::big_endian::get<decltype(out)>(data);
```
Comment (petya2164) [inline test/src/test_big_endian.cpp]: why do we need decltype here? just write float

File: test/src/test_big_endian.cpp [hunk 4]
```diff
@@ -208,4 +208,207 @@ TEST(test_big_endian, convert)
...
+        endian::big_endian::get_bytes<8>(out, data);
+        EXPECT_EQ(input, out);
+
+        out = endian::big_endian::get<decltype(out)>(data);
+        EXPECT_EQ(input, out);
+
+        out = endian::big_endian::get_bytes<sizeof(out), decltype(out)>(data);
+        EXPECT_EQ(input, out);
+
+        input = 0x1;
```
Comment (petya2164) [inline test/src/test_big_endian.cpp]: no hex notation, just 1.0

**Comment by petya2164** [comment]
I think this looks very useful. What do you think, @jpihl?

We should really split these "massive" tests to separate test cases per type. Here we should have ``convert_float`` and ``convert_double``. We could split the other types in another branch or just do it here.

**Comment by jpihl** [comment]
:+1:

**Pull request merged**